### PR TITLE
Add endpoints for managing group membership

### PR DESCRIPTION
**Pull request by rhafer**
- [x] POST /groups/{id}/members/$ref (for adding a single member)
 - [x] DELETE graph/v1.0/groups/{id}/members/{id}/$ref (for deleting a single member)
 - [x] PATCH graph/v1.0/groups/{group-id} (for adding multiple members)

### CODE REVIEW

**Comment by micbar** [comment]
shouldn't we be able to create a key `"members@odata.bind"` which has a collection of directory objects as a value?

**Review by micbar — APPROVED** [review]

**Comment by rhafer** [comment]
> shouldn't we be able to create a key `"members@odata.bind"`

@micbar hm, true. That should work. For some reason I thought the body for the PATCH on `/groups/{id}` required some additional attributes.Please review once more.

I still think the API for managing group memberships as defined in ms graph is not exactly nice. I am trying to follow it anyway here.

> which has a collection of directory objects as a value?

Not really directory objects, but strings/uris referencing directory objects.

**Review by refs — APPROVED** [review]

File: api/openapi-spec/v0.0.yaml [hunk 1]
```diff
@@ -498,6 +498,68 @@ paths:
         default:
           $ref: '#/components/responses/error'
       x-ms-docs-operation-type: operation
+  '/groups/{group-id}/members/$ref':
+    post:
+      tags:
+        - group
+      summary: Add a member to a group
+      operationId: AddGroupMember
```
Comment (refs) [inline api/openapi-spec/v0.0.yaml:506]: just a nitpick: I think `AddMember` reads better and could potentially lead to more concise code once this is generated into an sdk:

`group.AddGroupMember` vs `group.AddMember` kind of issue

File: api/openapi-spec/v0.0.yaml [hunk 2]
```diff
@@ -498,6 +498,68 @@ paths:
...
+          description: Success
+        default:
+          $ref: '#/components/responses/error'
+      x-ms-docs-operation-type: operation
+  '/groups/{group-id}/members/{directory-object-id}/$ref':
+    delete:
+      tags:
+        - group
+      summary: Delete member from a group
+      operationId: DeleteGroupMember
```
Comment (refs) [inline api/openapi-spec/v0.0.yaml:537]: same as the above comment

**Review by refs — APPROVED** [review]

**Comment by refs** [comment]
> shouldn't we be able to create a key `"members@odata.bind"` which has a collection of directory objects as a value?

@rhafer something like: https://swagger.io/docs/specification/data-models/data-types/#array

Also, mind this restriction by the msgraph spec:

> Note that up to 20 members can be added in a single request

The key would, obviously, be located in the `PATCH` request body

File: api/openapi-spec/v0.0.yaml (hunk 1 above)
Comment (rhafer) [reply api/openapi-spec/v0.0.yaml:506]: Makes sense. I'll update that and then merge.

**Comment by rhafer** [comment]
> > shouldn't we be able to create a key `"members@odata.bind"` which has a collection of directory objects as a value?
> 
> @rhafer something like: https://swagger.io/docs/specification/data-models/data-types/#array

Yeah that's what I did. :-)

> Also, mind this restriction by the msgraph spec:
> 
> > Note that up to 20 members can be added in a single request
> 
> The key would, obviously, be located in the `PATCH` request body

Yeah. Though that limit is not something I can express in this SPEC, is it? We might think about adding such a limit to the actual graph API implementation at some point though.

**Comment by refs** [comment]
yes it can be done within the spec: https://swagger.io/docs/specification/2-0/describing-parameters/

> Additionally, you can:
> 
>     use minItems and maxItems to control the size of the array,
>     enforce uniqueItems,

```yaml
- in: query
          name: color
          required: false
          type: array
          minItems: 1
          maxItems: 5
          uniqueItems: true
          items:
            type: string
            enum: [black, white, gray, red, pink, orange, yellow, green, blue, purple, brown]
```

**Comment by refs** [comment]
@rhafer  oh sorry, those docs are for v2. For v3: https://swagger.io/docs/specification/data-models/data-types/

image

**Comment by rhafer** [comment]
Ok, thanks. I've added those limits to the spec. It doesn't seem to have any effect on the generated go code for the client. But the limit itself needs to be implemented server side anyway.

**Pull request merged**